src/linter/block.go

```diff
@@ -1176,10 +1183,31 @@ func (b *BlockWalker) handleNew(e *expr.New) bool {
 		return true
 	}
 
-	if _, ok := meta.Info.GetClass(className); !ok {
+	info, ok := meta.Info.GetClass(className)
+	if !ok {
 		b.r.Report(e.Class, LevelError, "undefined", "Class not found %s", className)
 	}
 
+	// Check implicitly invoked constructor method arguments count.
+	ctor, ok := info.Methods["__construct"]
+	if !ok {
+		// TODO(quasilyte): we might want to catch case-insensitivity issues.
+		// Class may be called "Foo", but "foo" method can be its constructor.
+
+		// If constructor method uses className name instead of __construct.
```

Comment:
I think it is no longer supported in PHP 7.x so I would not even try to check it.